helper/schema/schema.go

```diff
@@ -91,8 +91,38 @@ type Schema struct {
 	// If CustomizeDiffFunc makes this field ForceNew=true, the
 	// following DiffSuppressFunc will come in with the value of old being
 	// empty, as if creating a new resource.
+	//
+	// By default, DiffSuppressFunc is considered only when deciding whether
+	// a configuration value is significantly different than the prior state
+	// value during planning. Set DiffSuppressAlways to make this rule take
```

Comment:
I believe you meant DiffSuppressOnRefresh here. (I'd use GitHub's suggestion feature but its harder to use to character limited lines.)

Reply:
Oh yes... funnily enough I did start off trying for an approach where `ResourceData.Set` would be the one to handle this and so had it called `DiffSuppressAlways` at that point, and missed this when I was switching approach. I'll update this.

---

I had meant to describe this in the writeup, but since you mentioned it in your comment above I'll add some notes about it here now for posterity:

There were two big challenges with doing this on `Set` which caused me to abandon that direction in favor of this one:
* `DiffSuppressFunc`'s signature is designed to deal with already-flatmapped data, because that's what's in the legacy diff data structure it was originally built to filter. `ResourceData` is built around the "field reader" abstraction which works with arbitrary `interface{}` and so I found myself having to essentially reimplement the flatmap functionality at that layer in order to get suitable inputs for the `DiffSuppressFunc` call.
* The `ResourceData.Set` API allows (and, in a lot of cases, _requires_) setting an entire nested data structure in a single call, which means visiting every leaf value in that data structure to decide whether to include it or not, and no part of the call stack in that area has all of the context necessary to make that decision.

While doing it after the fact against the `InstanceState` object is non-ideal, it benefits from more closely matching how the `Diff` method works with the `InstanceDiff` object as a post-processing step, and so it's a better fit for the existing `DiffSuppressFunc` design.